### PR TITLE
Replace some LiveData with Flow/StateFlow

### LINKED ISSUE #112: Migrate LiveData to StateFlow in ViewModels and observe in Fragments
**Is your feature request related to a problem? Please describe.**
To adapt Android modern development, we need to migrate from LiveData to StateFlow

**Describe the solution you'd like**
It would be great if we can just make 1 or some screens to be fully using Flow so that other developer can join and do it folowing the convention

**Describe alternatives you've considered**
None

### PR DESCRIPTION
**Pull request by DeKaN**
LiveData shouldn't be used at non-UI layers, I've replaced it with Flow
LiveData at UI layer (ViewModel) has been replaced to StateFlow

Fixes #112

### CODE REVIEW

File: app/src/main/java/com/hieuwu/groceriesstore/presentation/checkout/CheckOutFragment.kt [hunk 1]
```diff
@@ -61,16 +65,25 @@ class CheckOutFragment : Fragment() {
...
-        }
-
-        viewModel.totalPrice.observe(viewLifecycleOwner) {}
-        viewModel.isOrderSentSuccessful.observe(viewLifecycleOwner) {
-            if (it) {
-                openSuccessDialog()
-            } else {
-                showMessageSnackBar(getString(R.string.order_created_failed))
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
```
Comment (DeKaN) [inline app/src/main/java/com/hieuwu/groceriesstore/presentation/checkout/CheckOutFragment.kt:69]: Use this code if you want to collect more than one Flow

**Review by hieuwu — COMMENTED** [review]
It looks like this PR have conflicts with curent branch. Please help resolve it

**Comment by DeKaN** [comment]
@hieuwu Conflicts have been resolved

**Comment by hieuwu** [comment]
> @hieuwu Conflicts have been resolved

Cool. I am reviewing while the job is running

**Review by hieuwu — APPROVED** [review]
LGTM!

**Comment by Devendra34** [comment]
Hi @hieuwu,
I took a new pull after this PR merge. Looks like the cart feature is broken.
Steps to reproduce.
1. Launch the app and add some item from shop fragment using plus icon
2. open cart bottom sheet. (added items are shown properly till here)
3. now close this bottom sheet and add another item same as step 1
4. open the cart bottom sheet again. Now you will not see the recently added item in the list

This bug was not there before. The last stable merge was #128.
Can we please revert the changes or make the main branch stable until this Issue is fixed.
I have implemented #116, but due to this bug I have not pushed it yet. Let me know if its ok to push my code or should I wait.
Please check the video attached.
Thanks!

https://user-images.githubusercontent.com/51832211/193946013-965e3d7e-5c2a-48b6-88b2-f917232f622c.mp4

**Comment by hieuwu** [comment]
> 

Is this bug effect your work? If not, you can push yours, I will create another issue for this bug. Thanks for pointing this out @Devendra34

**Comment by hieuwu** [comment]
@all-contributors please add @DeKaN for code

**Pull request merged**